File: multicast_core/tests/src/sender.cpp
#include <multicast_core.h>

#include <chrono>
#include <iostream>
#include <opencv2/opencv.hpp>

#define MULTICAST_GRP "224.0.0.1"
#define MULTICAST_PORT 5000

int main() {
    MulticastLib::Sender s(MULTICAST_GRP, MULTICAST_PORT);

    // Запуск стрима
    if (!s.startStream()) {
        std::cerr << "failed to start stream" << std::endl;
        return -1;
    }

    while (1) {
        // Отображение превью
        cv::Mat frame = s.getPreviewFrame();
        if (frame.empty()) {
            std::cerr << "Empty Frame" << std::endl;
            continue;
        }
        cv::imshow("SEND PREVIEW CAM", frame);

        // Выход по нажатию ESC
        if (cv::waitKey(1) == 27) {  // 27 — ESC
            break;
        }
    }

    s.stopStream();
    return 0;
}

File: gui/sender.py
import time

import cv2

import multicast_core

MCAST_GRP = "224.0.0.1"
MCAST_PORT = 5000


def main():
    sender = multicast_core.Sender(MCAST_GRP, MCAST_PORT)

    if not sender.start_stream():
        print("Failed to start stream")
        return

    try:
        while True:
            frame = sender.get_preview_frame()
            if frame is None or frame.size == 0:
                print("No valid frame")
                time.sleep(0.1)
                continue

            cv2.imshow("Sender Preview", frame)
            if cv2.waitKey(1) & 0xFF == 27:
                break
    except KeyboardInterrupt:
        print("Stream interrupted by user")
    finally:
        sender.stop_stream()
        cv2.destroyAllWindows()


if __name__ == "__main__":
    main()
